### PR TITLE
fix: generate csv for disperse and gnosis (#15)

### PR DESCRIPTION
**Pull request by aahna-ashina**
- [x] https://github.com/nation3/nationcred-instance/pull/28/commits/ae94e8632c556ce658e6be516c27ed27194013d7 Write custom script for generating CSVs compatible with Disperse.app
- [x] https://github.com/nation3/nationcred-instance/pull/28/commits/379629549c3333bfb1965ffd466383097562d699 Write custom script for generating CSVs compatible with Gnosis Safe
  - [x] https://github.com/nation3/nationcred-instance/pull/28/commits/c4ea2e63c2f0239a9f1013f304a5e532278ef2e5 Include column headers in Gnosis CSV

### CODE REVIEW

**Comment by aahna-ashina** [comment]
@johnmark13 PR: https://github.com/nation3/nationcred-instance/pull/28/files

**Review by johnmark13 — APPROVED** [review]
Nice work, I'd put a note on the padding code as to source and desitnation units, but as long as that all works, you're all good. I guess you need it this week too...

File: generate-csv.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ *   0x3e465ABFa9b2A7E18a610F489fb3510765461d13,0.007718330904890492
+ */
+function convertAmountFormat(csvRows) {
+    console.log('convertAmountFormat')
+
+    csvRows.forEach(function(row, index) {
+        // Prepend zeros
+        while (row.amount.length <= 18) {
+            row.amount = "0" + row.amount
+        }
```
[the remark this answers is not in the document]
Comment (aahna-ashina) [reply generate-csv.js:70]: @johnmark13 I tried changing to `row.amount = row.amount.padStart(18, '0')`, but it resulted in a missing zero in front of the decimal:
```
0x540A6950376edC97D449EcCfd1987C100843Be72,335847891573817
```
    ↓
```csv
0x540A6950376edC97D449EcCfd1987C100843Be72,.000335847891573817
```

[the remark this answers is not in the document]
Comment (aahna-ashina) [reply generate-csv.js:70]: Should be `0.000335847891573817`

File: generate-csv.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+function writeToGnosisCSV(filePathGnosis, csvRows) {
+    console.log('writeToGnosisCSV')
+
+    // Add missing columns
+    csvRows.forEach(function(row, index) {
+        row.token_type = 'erc20'
+        row.token_address = '0x333A4823466879eeF910A04D473505da62142069'
+    })
```
[the remark this answers is not in the document]
Comment (aahna-ashina) [reply generate-csv.js:95]: @johnmark13 Yes, agree 100%. If we end up actually using this script in the long term, let's implement this 👍

**Comment by aahna-ashina** [comment]
@johnmark13 You can see sample output of the current version of this script at https://github.com/nation3/nationcred-instance/pull/29/commits/b0f4b12c36f625d1405ffc139b1d4db975fa2307

File: generate-csv.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (aahna-ashina) [reply generate-csv.js:70]: @johnmark13 All right, let's do some refactoring later, unless #30 makes this entire script obsolete.

**Pull request merged**